eno-ws/src/main/java/fr/insee/eno/ws/controller/GenerationController.java

```diff
@@ -0,0 +1,151 @@
+package fr.insee.eno.ws.controller;
+
+import fr.insee.eno.legacy.parameters.*;
+import fr.insee.eno.ws.PassePlat;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.MediaType;
+import org.springframework.http.codec.multipart.FilePart;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.http.server.reactive.ServerHttpResponse;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import reactor.core.publisher.Mono;
+
+@Tag(name = "Generation of questionnaire")
+@Controller
+@RequestMapping("/questionnaire")
+@Slf4j
+@SuppressWarnings("unused")
+public class GenerationController {
+	
+	private final PassePlat passePlat;
+
+	public GenerationController(PassePlat passePlat) {
+		this.passePlat = passePlat;
+	}
+
+	@Operation(
+			summary = "Generation of questionnaire according to parameters.",
+			description = "Generate a questionnaire using the parameters file (required), metadata file (optional) " +
+					"and the specificTreatment file (optional). To use it, you have to upload all necessary files.")
+	@PostMapping(value = "in-2-out", 
+			produces = MediaType.APPLICATION_OCTET_STREAM_VALUE, consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+	public Mono<Void> generate(
+			@RequestPart(value="in", required=true) Mono<FilePart> in,
+			@RequestPart(value="params", required=true) Mono<FilePart> params,
+			@RequestPart(value="metadata", required=false) Mono<FilePart> metadata,
+			@RequestPart(value="specificTreatment", required=false) Mono<FilePart> specificTreatment,
+			@RequestPart(value="mapping", required=false) Mono<FilePart> mapping,
+			@RequestParam(value="multi-model", required=false, defaultValue="false") boolean multiModel,
+			ServerHttpRequest request, ServerHttpResponse response) throws Exception {
+		return passePlat.passePlatPost(request, response);
+	}
+
+	@Operation(
+			summary = "Generation of XSL-FO questionnaire.",
+			description = "Generate a XSL-FO questionnaire from a DDI questionnaire using the FO parameters given.")
+	@PostMapping(value = "ddi-2-fo",
+			produces = MediaType.APPLICATION_OCTET_STREAM_VALUE, consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+	public Mono<Void> generateFOQuestionnaire(
```

Comment:
is there an interest in having params validation here ? It could be error prone in some cases to have validation here and on the underlying webservice